apollo-router-core/src/plugins/csrf.rs

```diff
@@ -0,0 +1,320 @@
+use crate::{register_plugin, Plugin, RouterRequest, RouterResponse, ServiceBuilderExt};
+use http::header;
+use http::{HeaderMap, StatusCode};
+use schemars::JsonSchema;
+use serde::Deserialize;
+use std::ops::ControlFlow;
+use tower::util::BoxService;
+use tower::{BoxError, ServiceBuilder, ServiceExt};
+
+#[derive(Deserialize, Debug, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+struct CSRFConfig {
+    /// CSRFConfig is enabled by default;
+    /// set disabled = true to disable the plugin behavior
+    #[serde(default)]
+    disabled: bool,
+    /// Override the headers to check for by setting
```

Comment:
CSRF prevention = block clients (most notably GET requests) that don't jump through certain hoops. At the very least, this might require a bit of work for users when they upgrade to this version for the first time.